### ISSUE TITLE
pusa wan video

### ISSUE THREAD

**Issue by Rudra-ai-coder** [kijai/ComfyUI-WanVideoWrapper#804]
can we have a lora extraction for this @kijai https://huggingface.co/RaphaelLiu/PusaV1

**Comment by kijai** [comment]
I did that earlier, but haven't been able to get any useful output with it.

**Comment by Maki9009** [comment]
why lora? why not as a replacement model rather than wan? i though that was the whole point of this model

**Comment by jepjoo** [comment]
> why lora? why not as a replacement model rather than wan? i though that was the whole point of this model

PusaV1 is finetuned from Wan2.1 T2V. Typically it would then be possible to extract the difference between Wan2.1 T2V and PusaV1 into a LoRA.

**Comment by kijai** [comment]
I'm very confused over this release, the weights they shared are just the normal Wan 14B (identical outputs). Then there's the .pt file that's split into multiple files, that can be joined and loaded and it includes training data, checkpoint and lora weights. The checkpoint weights again seems almost identical to base Wan 14B T2V, and when loading the LoRA weights to any model, normal or this included one, it just produces bad outputs even with the code changes they have in their repo.

In their example code a normal Wan model is loaded, then a lora. I'm either missing something obvious or this release isn't finished yet.

**Comment by Maki9009** [comment]
I thought the .pt files are is a model that does it all in 10 steps... basically i2v model that also does all the capabilities of VACE models I think and also does t2v without breaking it

**Comment by mr-lab** [comment]
maybe this can shed some light 

> I did that earlier, but haven't been able to get any useful output with it.


https://github.com/Yaofang-Liu/Pusa-VidGen/issues/4#issuecomment-2802065657
> Hi, thanks for the details. **Since this is an early preview version, the results sometimes do collapse.** But for most cases, if you try different `cond_position` (from 0-27) and `noise_multiplier` (from 0-1), you're likely to get better results. For example, if you use `cond_position=21`, `noise_multiplier=0.4`, you can get this result:
> 
>  play_basketball_20250414_215655.mp4 
> Note that I updated the script for I2V for you to control the noise added to the condition image more easily by setting this `noise_multiplier` parameter. And I also add an example image. You will get result for this example image like:
> 
>  00034_20250413_193853.mp4 
> Please ask me if you have any further quesitons. Thanks!

**Comment by Yaofang-Liu** [comment]
> I'm very confused over this release, the weights they shared are just the normal Wan 14B (identical outputs). Then there's the .pt file that's split into multiple files, that can be joined and loaded and it includes training data, checkpoint and lora weights. The checkpoint weights again seems almost identical to base Wan 14B T2V, and when loading the LoRA weights to any model, normal or this included one, it just produces bad outputs even with the code changes they have in their repo.
> 
> In their example code a normal Wan model is loaded, then a lora. I'm either missing something obvious or this release isn't finished yet.

Hi, thanks for your effort on PusaV1. Actually the model is truly a lora with lora rank 512 (about 2B parameters trained). We use diffsynth-studio for implementation, and it automatically saves it as a whole .pt file as large as the base model. I would like to look into it and see if I can split out the Lora part. Besides, by bad outputs, do you mean the I2V quality is not good, or it just not working or collapse? Did you reproduce the similar results like the demos on our project page? Please let me know if there is any further problems :)

**Comment by kijai** [comment]
> > I'm very confused over this release, the weights they shared are just the normal Wan 14B (identical outputs). Then there's the .pt file that's split into multiple files, that can be joined and loaded and it includes training data, checkpoint and lora weights. The checkpoint weights again seems almost identical to base Wan 14B T2V, and when loading the LoRA weights to any model, normal or this included one, it just produces bad outputs even with the code changes they have in their repo.
> > In their example code a normal Wan model is loaded, then a lora. I'm either missing something obvious or this release isn't finished yet.
> 
> Hi, thanks for your effort on PusaV1. Actually the model is truly a lora with lora rank 512 (about 2B parameters trained). We use diffsynth-studio for implementation, and it automatically saves it as a whole .pt file as large as the base model. I would like to look into it and see if I can split out the Lora part. Besides, by bad outputs, do you mean the I2V quality is not good, or it just not working or collapse? Did you reproduce the similar results like the demos on our project page? Please let me know if there is any further problems :)

I think there's a mistake in your example regarding T2V which initially threw me off, [the first timestep](https://github.com/Yaofang-Liu/Pusa-VidGen/blob/812ba87d889a41acd2001d9714b6df3c330cfc7b/PusaV1/diffsynth/pipelines/wan_video_pusa.py#L469) is always zeroed so it wouldn't work and that caused the confusion. 

I realized this only when trying I2V with more start frames, so now I understand how to use this.

I've extracted the lora layers to a Comfy compatible file here: 

https://huggingface.co/Kijai/WanVideo_comfy/blob/main/Pusa/Wan21_PusaV1_LoRA_14B_rank512_bf16.safetensors

And added basic support for the inference, still work in progress and I seem to have quality issues I haven't figured out yet.

example with 13 start frames:

https://github.com/user-attachments/assets/ccf56c9e-d84d-458a-8e48-1f0d3e6d9393

**Comment by Yaofang-Liu** [comment]
> > > I'm very confused over this release, the weights they shared are just the normal Wan 14B (identical outputs). Then there's the .pt file that's split into multiple files, that can be joined and loaded and it includes training data, checkpoint and lora weights. The checkpoint weights again seems almost identical to base Wan 14B T2V, and when loading the LoRA weights to any model, normal or this included one, it just produces bad outputs even with the code changes they have in their repo.
> > > In their example code a normal Wan model is loaded, then a lora. I'm either missing something obvious or this release isn't finished yet.
> > 
> > 
> > Hi, thanks for your effort on PusaV1. Actually the model is truly a lora with lora rank 512 (about 2B parameters trained). We use diffsynth-studio for implementation, and it automatically saves it as a whole .pt file as large as the base model. I would like to look into it and see if I can split out the Lora part. Besides, by bad outputs, do you mean the I2V quality is not good, or it just not working or collapse? Did you reproduce the similar results like the demos on our project page? Please let me know if there is any further problems :)
> 
> I think there's a mistake in your example regarding T2V which initially threw me off, [the first timestep](https://github.com/Yaofang-Liu/Pusa-VidGen/blob/812ba87d889a41acd2001d9714b6df3c330cfc7b/PusaV1/diffsynth/pipelines/wan_video_pusa.py#L469) is always zeroed so it wouldn't work and that caused the confusion.
> 
> I realized this only when trying I2V with more start frames, so now I understand how to use this.
> 
> I've extracted the lora layers to a Comfy compatible file here:
> 
> https://huggingface.co/Kijai/WanVideo_comfy/blob/main/Pusa/Wan21_PusaV1_LoRA_14B_rank512_bf16.safetensors
> 
> And added basic support for the inference, still work in progress and I seem to have quality issues I haven't figured out yet.
> 
> example with 13 start frames:
> 
>  WanVideoWrapper_I2V_00189.1.mp4

You're right, sorry for the bugs and my oversight, and I really appreciate your patience and support!! That helps a lot 👍  

Please check the newest version that I basically fix all the bugs and tested for multiple runs to get some showcases for each example (you shall see on the [README.md](https://github.com/Yaofang-Liu/Pusa-VidGen/tree/main/PusaV1)). For your example, I also tried to crop and resize it to 720p, but seems that your example is some kind of low quality, do you tried other high quality examples? Please let me know if there is any further problems. Besides, I will also recommend the Lora checkpoint and Comfy stuffs to our README. Many thanks! :)

**Comment by manzonif** [comment]
Somehow it seems to work pretty well with the Vace module, (withowt the Lightx2v lora). In 10 steps I found great consistency to the ref image. I tested only with depth.

 I found also that it require more VRAM. With only the Vace module I'am able to render up at 1008x1008px x 81 frames. With Pusa + Vace and flowmatch_pusa scheduler, only 816 x816x81.

**Comment by kijai** [comment]
Some examples of better quality outputs, these are with the Lightx2v distill LoRA and cfg 1.0:

https://github.com/user-attachments/assets/229d5c99-26a0-47f9-87e7-f796beae3bfe

https://github.com/user-attachments/assets/bc9bbefc-0210-4b43-b078-6e602bc3f464

https://github.com/user-attachments/assets/9d53bc3c-7d60-48ad-9d35-b5fbf8c040a8

**Comment by ameen-roayan** [comment]
@kijai i know we are still in the figuring it out phase, but in an attempt to follow i tried the updated workflow from yesterday but was getting "WanVideoEmptyEmbeds.process() got an unexpected keyword argument 'extra_latents'"

**Comment by kijai** [comment]
> @kijai i know we are still in the figuring it out phase, but in an attempt to follow i tried the updated workflow from yesterday but was getting "WanVideoEmptyEmbeds.process() got an unexpected keyword argument 'extra_latents'" 

Update and/or refresh the node, I changed the name of the extra latents at one point.

**Comment by WainWong** [comment]
Thank you for your work. I have a few more questions I would like to ask. I noticed that Pusa officially claims to support frame padding, frame completion, and other features. I noticed that your current workflow demonstration only includes one image-to-video conversion. Are you still developing the other features, or can these features be implemented using the previous workflow?

**Comment by a3nima** [comment]
> > [@kijai](https://github.com/kijai) i know we are still in the figuring it out phase, but in an attempt to follow i tried the updated workflow from yesterday but was getting "WanVideoEmptyEmbeds.process() got an unexpected keyword argument 'extra_latents'"
> 
> Update and/or refresh the node, I changed the name of the extra latents at one point.

refreshing the node fixes the problem. but maybe related to this, with the example workflow only the text prompt seems to be honored not the input image.

**Comment by Os-De** [comment]
I am very new to this, am facing this issue <img width="1706" height="661" alt="Image" src="https://github.com/user-attachments/assets/6c5b2718-50db-4d70-8b76-26535ff138b9" /> still have no idea how to put the flowmatch_pusa in the WanVideo Sampler

**Comment by kijai** [comment]
> I am very new to this, am facing this issue <img alt="Image" width="1706" height="661" src="https://private-user-images.githubusercontent.com/164570299/468764991-6c5b2718-50db-4d70-8b76-26535ff138b9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMxMjYyNzQsIm5iZiI6MTc1MzEyNTk3NCwicGF0aCI6Ii8xNjQ1NzAyOTkvNDY4NzY0OTkxLTZjNWIyNzE4LTUwZGItNGQ3MC04Yjc2LTI2NTM1ZmYxMzhiOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcyMVQxOTI2MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zOTQ0NjYzYTRiYzA3YTA0YTZlNGMwY2VmZTlkNmMxNWU1NzZiNDRiZjcxMjY5NWZlMzFhYmM2MjM0YWI1Zjk2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ZxrpzjOip-xnTfAr85PBrZB-WpWnqC7pqFrYY39lZrE"> still have no idea how to put the flowmatch_pusa in the WanVideo Sampler

You update the nodes and refresh your browser, not much else to it... if it doesn't show up when you do that, then the nodes are not updating for some reason, easiest just to re-install then.

**Comment by Yaofang-Liu** [comment]
> Some examples of better quality outputs, these are with the Lightx2v distill LoRA and cfg 1.0:
> 
>  WanVideoWrapper_I2V_00215.mp4 
>  WanVideoWrapper_I2V_00201.mp4 
>  WanVideoWrapper_I2V_00240.mp4

Thank you kijai! i am wondering do we really need to use Lightx2v? since **5 steps** can already achieve good results with Pusa, do not need to add other components, we just use 10 steps for SOTA, you may see more details from the [report](https://github.com/Yaofang-Liu/Pusa-VidGen/blob/e99c3dcf866789a2db7fbe2686888ec398076a82/PusaV1/PusaV1.0_Report.pdf)
<img width="717" height="94" alt="Image" src="https://github.com/user-attachments/assets/9ba3fbd8-4280-4dc9-a966-d944223172ad" />

**Comment by kijai** [comment]
> > Some examples of better quality outputs, these are with the Lightx2v distill LoRA and cfg 1.0:
> > WanVideoWrapper_I2V_00215.mp4
> > WanVideoWrapper_I2V_00201.mp4
> > WanVideoWrapper_I2V_00240.mp4
> 
> Thank you kijai! i am wondering do we really need to use Lightx2v? since **5 steps** can already achieve good results with Pusa, do not need to add other components, we just use 10 steps for SOTA, you may see more details from the [report](https://github.com/Yaofang-Liu/Pusa-VidGen/blob/e99c3dcf866789a2db7fbe2686888ec398076a82/PusaV1/PusaV1.0_Report.pdf) <img alt="Image" width="717" height="94" src="https://private-user-images.githubusercontent.com/45255738/469369221-9ba3fbd8-4280-4dc9-a966-d944223172ad.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMyODMwNjcsIm5iZiI6MTc1MzI4Mjc2NywicGF0aCI6Ii80NTI1NTczOC80NjkzNjkyMjEtOWJhM2ZiZDgtNDI4MC00ZGM5LWE5NjYtZDk0NDIyMzE3MmFkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzIzVDE0NTkyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU3MDc4MDNhYzFlOTNmZjZiNzQ2M2UxYTU3NGE2MjdkZDBjNTE2YWVkZDY3YTIzYzU1NjUwNDJkMmUzMmM0YWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ymXE9ekmDHtiB-Pa45i0ZRIlUsefYEk-V8DgDgZMRc8">

I wonder if I'm doing something wrong then, I've tried with both the LoRA I initially took from the huge .pt file and the one you posted on huggingface later, both give identical outputs and at 5 steps with cfg 5.0 it basically doesn't work at all. If I add Lightx2v it works very well with cfg 1.0.

**Comment by Yaofang-Liu** [comment]
Some results could be very bad, I think your configuration should be right. here are some results I tested with 5 steps, cfg 5.0, zero noise for the first frame:

https://github.com/user-attachments/assets/c1ff1c3c-e2ad-441a-b1db-a44b654617d3

https://github.com/user-attachments/assets/e093d6be-b422-4f0e-b373-841d6483dfa0

https://github.com/user-attachments/assets/b7eab00e-d776-41bf-9a5e-20bcc4ecf1a3

Lower quality than 10 steps (especially for the first official demo on [README](https://github.com/Yaofang-Liu/Pusa-VidGen/tree/main/PusaV1)), but I don't know if it's worse than that with Lightx2v. If so, Lightx2v may be truly useful

**Comment by kijai** [comment]
Ok well that's about what I get too, but when adding lightx2v LoRA at 0.7 strength and using Pusa LoRA at 1.5 strength I can get this in 5 steps with cfg 1.0, so twice as fast:

https://github.com/user-attachments/assets/927f61c2-b4a3-4272-950e-b2700842bd6a

https://github.com/user-attachments/assets/4dffbf52-5ee6-4867-8db9-023c327cdbcb

**Comment by Yaofang-Liu** [comment]
Impressive. I'd like to look into its mechanism. Many thanks!

**Comment by Heliumrich** [comment]
I tried CausVid + Lightx2v with CFG 1 and had barely any movement.
Was it caused by causvid, and not an issue here ?

We got so many wan2.x loras to improve/speedup it, it's hard to know which one overlap or have compat issues ...

BTW anyone tried pusa1 with wan2.2 ?

**Comment by Rudra-ai-coder** [comment]
wan 2.2 inbuilt has high quality i would not recommend pusa 2 on that

**Comment by a3nima** [comment]
> BTW anyone tried pusa1 with wan2.2 ?

I've tried it with 10 steps and it worked. But havent made quality comparisons.

**Comment by Tinyyuumi** [comment]
> > BTW anyone tried pusa1 with wan2.2 ?
> 
> I've tried it with 10 steps and it worked. But havent made quality comparisons.

wan2.2 has two models: high-noise and low-noise. How do you load lora?
